UXW-2576 show tenant collections in the trash

johnswanson opened the pull request.

johnswanson committed:
UXW-2576 show tenant collections in the trash

I also deleted the `audit-namespace-clause` function. It felt like it
didn't do enough to justify its existence - every caller needed to
modify the list of namespaces. Could probably DRY things up a bit still,
but think we should change the approach.

edpaget commented on src/metabase/collections_rest/api.clj:
```diff
@@ -707,7 +707,17 @@
+                        [[:= :namespace nil]
+                         (when (premium-features/enable-audit-app?)
+                           [:= :namespace "analytics"])
+                         (when (collection/is-trash? collection)
```
do we need to check for the feature flag here?

johnswanson commented on src/metabase/permissions/models/collection/graph.clj:
```diff
@@ -110,7 +111,11 @@
                      :where [:and
                              [:or [:= :type nil] [:not= :type [:inline "trash"]]]
-                             (perms/audit-namespace-clause :namespace (u/qualified-name collection-namespace))
+                             (if-let [namespace (u/qualified-name collection-namespace)]
```
Yea I'm not super tied to removing it, it just felt so silly to be adding extra conditions to every place we called it?

johnswanson committed:
Check `use-tenants` before returning tenant colls

johnswanson committed:
s/audit-namespace-clause/namespace-clause helper

`namespace-clause` takes the ns column, ns value, and an optional
argument for whether *this* particular result should include the tenant
collections in the response (since sometimes we want 'em, sometimes we
don't.)

johnswanson committed:
kondo
